### PR TITLE
[SPARK-16395] [STREAMING] Fail if too many CheckpointWriteHandlers are queued up in the fixed thread pool

### PR DESCRIPTION
**Pull request by srowen**
## What changes were proposed in this pull request?

Begin failing if checkpoint writes will likely keep up with storage's ability to write them, to fail fast instead of slowly filling memory
## How was this patch tested?

Jenkins tests

### CODE REVIEW

**Comment by srowen** [comment]
CC @tdas

File: streaming/src/main/scala/org/apache/spark/streaming/Checkpoint.scala [hunk 1]
```diff
@@ -18,8 +18,8 @@
 package org.apache.spark.streaming
 
 import java.io._
-import java.util.concurrent.Executors
-import java.util.concurrent.RejectedExecutionException
+import java.util.concurrent.{ArrayBlockingQueue, RejectedExecutionException,
```
Comment (tdas) [inline streaming/src/main/scala/org/apache/spark/streaming/Checkpoint.scala]: do we usually break up the import lines this way? i dont think so, but I may be wrong.

Comment (srowen) [inline streaming/src/main/scala/org/apache/spark/streaming/Checkpoint.scala]: Yeah it's the style I see elsewhere, like https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/SparkContext.scala#L39  Arguably it's time for a _ import at this stage; I'm indifferent here.

**Comment by srowen** [comment]
@tdas how does the substance of the change strike you? it's an improvement over the current situation where millions of checkpoints would stack up, but the cap of 1000 was arbitrary. I didn't necessarily think it worth making configurable, just "large", but YMMV

**Comment by srowen** [comment]
I'm gonna go for this if there are no objections. I'd even put it into 2.0, but unless someone seconds that I won't.

**Pull request closed without merging**